src/main/java/slimeknights/tconstruct/gadgets/item/FlintShurikenItem.java

```diff
@@ -0,0 +1,34 @@
+package slimeknights.tconstruct.gadgets.item;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.stats.Stats;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.gadgets.entity.ThrowableItem;
+import slimeknights.tconstruct.gadgets.entity.shuriken.FlintShurikenEntity;
+
+public class FlintShurikenItem extends ThrowableItem {
+
+  @Override
+  public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn) {
+    ItemStack itemStack = playerIn.getHeldItem(handIn);
+    if (!playerIn.abilities.isCreativeMode) {
+      itemStack.shrink(1);
+    }
+
+    playerIn.getCooldownTracker().setCooldown(itemStack.getItem(), 4);
+
+    if(!worldIn.isRemote) {
+      FlintShurikenEntity entity = new FlintShurikenEntity(worldIn, playerIn);
```

Comment:
This line is the only difference between the two classes. I suggest making one `ShurikenItem` class with a `BiFunction<ShurikenEntity,World,PlayerEntity> entity` parameter. Will let you pass `FlintShurikenEntity::new` into the constructor and call the function here.